src/util.rs

```diff
@@ -183,20 +187,42 @@ where
     }
 }
 
-#[allow(dead_code)]
-pub fn convert_vec_f32(mut a: Vec<u8>, e: Endianness) -> Vec<f32> {
-    if e != Endianness::system() {
-        for c in a.chunks_mut(4) {
-            let (a, b) = c.split_at_mut(2);
-            mem::swap(&mut a[0], &mut b[1]);
-            mem::swap(&mut a[1], &mut b[0]);
+#[cfg(feature = "ndarray_volumes")]
+pub fn convert_bytes_and_cast_to<I, O>(
+    raw_data: Vec<u8>,
+    e: Endianness,
+    dim: &Vec<Ix>,
+    slope: O,
+    inter: O
+    ) -> Array<O, IxDyn>
+    where I: AsPrim + PodTransmutable,
+          O: AsPrim + Num
+{
+    let data: Vec<I> = convert_bytes_to(raw_data, e);
+    Array::from_shape_vec(IxDyn(&dim).f(), data)
+        .expect("Inconsistent raw data size")
+        .mapv(|v| raw_to_value::<I, O>(v, slope, inter))
+}
+
+#[cfg(feature = "ndarray_volumes")]
+pub fn convert_bytes_to<T: PodTransmutable>(
+    mut a: Vec<u8>,
+    e: Endianness
+) -> Vec<T> {
+    let nb_bytes = mem::size_of::<T>();
+    if e != Endianness::system() && nb_bytes > 1 {
+        // Swap endianness by block of nb_bytes
+        let split_at = nb_bytes / 2;
+        let split_at_m1 = split_at - 1;
+        for c in a.chunks_mut(nb_bytes) {
+            let (a, b) = c.split_at_mut(split_at);
+            for i in 0..split_at {
+                mem::swap(&mut a[i], &mut b[split_at_m1 - i]);
+            }
         }
     }
-    let mut v: Vec<f32> = guarded_transmute_pod_vec_permissive(a);
-    for e in &mut v {
-        *e = designalise_f32(*e);
```

Comment:
It feels problematic to leave this transformation out, but I'll seek to solve this before the next release.

Reply:
I wasn't sure when I removed it because I wonder if modifying the input data is "right". I mean, there's probably a reason why there's a Signaling NaN in an image. I don't know of any, but why would it be there?

A better question might be "Why did you add the call to `designalise_f32`?"

Reply:
Usually, there isn't a signalling NaN. But a file having one is not impossible. This, along with the fact that the Rust community at large has preferred not to take any chances, is what drives me to still convert all NaNs to quiet NaNs.

As the number of data types possible is pretty limited, one can build efficient and safe implementations for reading one or more values without too much boilerplate. This will also make dealing with these corner cases possible without exposing traits related to implementation details. My current concern is that I have never found an RGB or Complex NIfTI-1 volume in practice. If you happen to know more on the subject, I would be glad to talk about it on issue #3. :)